### PR TITLE
Add mirror variant

### PR DESCRIPTION
**Pull request by kstinson14**
In order to shorten the process for mirroring css attributes for RTL, I made the `mirror` variant.  Doing something like `mirror:text-left`, will give the element left text alignment in left-to-right components, and right text alignment in right-to-left components.

I thought about just extending the `rtl` variant to have this behavior, but that would make designing an rtl-first process, which seemed a bit unintuitive.

### CODE REVIEW

**Comment by jrowlingson** [comment]
> I thought about just extending the `rtl` variant to have this behavior, but that would make designing an rtl-first process, which seemed a bit unintuitive.

Could you expand a bit on this?

File: tailwind.config.js [hunk 1]
```diff
@@ -235,19 +235,25 @@ module.exports = {
 
   variants: {
     backgroundColor: [ 'responsive', 'rtl', 'hover', 'focus' ],
+    backgroundPosition: [ 'responsive', 'mirror' ],
+    borderColor: [ 'dark', 'hover', 'focus' ],
+    borderRadius: [ 'responsive', 'mirror' ],
```
Comment (jrowlingson) [inline tailwind.config.js:237]: Is the `mirror` variant needed here?

Comment (kstinson14) [reply tailwind.config.js:237]: Yeah, I think we do want it for border-radius.  The first usage that jumps to my mind is in the chip component for the "X" button:

Screen Shot 2020-05-28 at 11 01 10 AM

Only one side is rounded, and it changes based on the "rtl" status

**Comment by kstinson14** [comment]
@jrowlingson Yeah, I didn't explain that very well.  I think I should have said "coding" would become an rtl-first process rather than "designing".  

Basically, I first thought of just making the`rtl` variant automatically reverse the utility for "LTR" layouts.  For example, using the class `rtl:ml-4` would result in a left margin of 1rem in RTL layouts, and a right margin of 1rem in LTR layouts.  But this would mean adding classes would involve listing out all of the RTL styling, instead of the LTR styling (assuming this variant works the same way we've been using it in our apps, and rtl:[some-class] means that [some-class] is applied when `dir="rtl"`).   Since we design for LTR layouts,  having all the classes pertain to the RTL layout seems unintuitive.

File: tailwind.config.js [hunk 2]
```diff
@@ -272,7 +278,48 @@ module.exports = {
     function({ addVariant, e }) {
       addVariant('rtl', ({ modifySelectors, separator }) => {
         modifySelectors(({ className }) => {
-          return `[dir="rtl"] .${e(`rtl${separator}${className}`)}`
+          return `:host([dir="rtl"]) .${e(`rtl${separator}${className}`)}`
+        })
+      })
+      addVariant('mirror', ({ container, separator }) => {
+        const newRules = []
+        container.each(rule => {
```
Comment (jrowlingson) [inline tailwind.config.js]: walkRules?

File: tailwind.config.js [hunk 3]
```diff
@@ -272,7 +278,48 @@ module.exports = {
...
+                  ? node.value.replace('left', 'right')
+                  : node.value.replace('right', 'left')
+              }
+            })
+            newRules.push(clone)
+          }
+        })
+        container.append(newRules)
+      })
+      addVariant('dark', ({ modifySelectors, separator }) => {
```
Comment (jrowlingson) [inline tailwind.config.js]: Remove?

File: tailwind.config.js [hunk 4]
```diff
@@ -235,19 +235,25 @@ module.exports = {
 
   variants: {
     backgroundColor: [ 'responsive', 'rtl', 'hover', 'focus' ],
+    backgroundPosition: [ 'responsive', 'mirror' ],
+    borderColor: [ 'dark', 'hover', 'focus' ],
```
Comment (jrowlingson) [inline tailwind.config.js]: Remove?

**Pull request closed without merging**